swap/src/protocol.rs

```diff
@@ -75,7 +75,7 @@ pub struct Message4 {
     tx_cancel_sig: bitcoin::Signature,
 }
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
```

Comment:
Why do these suddenly have to be serializable?

Reply:
Ah I see, you are not using the same structs as you did for the sled DB. We already have them, I think we can just use them across both implementations.

Making these types serializable brings back the issue of easily making breaking changes across two versions because you cannot deserialize the old state from the DB.